pkg/webui/console/components/duration-input/index.js

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import classnames from 'classnames'
+import bind from 'autobind-decorator'
+
+import Select from '../../../components/select'
+import Input from '../../../components/input'
+import { unit as unitRegexp } from '../../lib/regexp'
+import PropTypes from '../../../lib/prop-types'
+import style from './duration-input.styl'
+
+class DurationInput extends React.PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    decode: PropTypes.func,
+    encode: PropTypes.func,
+    name: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired,
+    units: PropTypes.arrayOf(
+      PropTypes.shape({
+        label: PropTypes.message,
+        value: PropTypes.string,
+      }),
+    ).isRequired,
+    value: PropTypes.string,
+  }
+
+  static defaultProps = {
+    className: undefined,
+    encode: (duration, unit) => (duration ? `${duration}${unit}` : ''),
+    decode: value => {
+      const duration = value.split(unitRegexp)[0]
+      const unit = value.split(duration)[1]
+      return {
+        duration: Number(duration),
+        unit,
+      }
+    },
+    value: undefined,
+  }
+
+  constructor(props) {
+    super(props)
+    const { value, decode } = this.props
+
+    const initialDelay = decode(value)
+    this.state = {
```

Comment:
You don't need to keep state of the value. The value prop passed to the component is sufficient and you can derive all the values you need from it. I propose to use the `withComputedProps` HOC, like this:
```js
@withComputedProps(props => ({
  ...props,
  value: props.decode(props.value),
}))
class DurationInput extends React.PureComponent {
   …
```
This way, the value prop will always be in the form that you need and you can encode it again before passing it to the `onChange` prop.

Since the component is quite simple and only uses the value prop in the render function, you could alternatively just pipe the value through the decoder function there.